x-pack/plugins/ingest_manager/server/routes/agent/upgrade_handler.ts

```diff
@@ -26,14 +27,18 @@ export const postAgentUpgradeHandler: RequestHandler<
 > = async (context, request, response) => {
   const soClient = context.core.savedObjects.client;
   const { version, source_uri: sourceUri } = request.body;
-
+  // get version number only in case "-SNAPSHOT" is in it
+  const kibanaVersionNumber = semver.coerce(appContextService.getKibanaVersion())?.version;
+  if (!kibanaVersionNumber) throw new Error(`kibanaVersion ${kibanaVersionNumber} is not valid`);
+  const versionToUpgradeNumber = semver.coerce(version)?.version;
+  if (!versionToUpgradeNumber)
+    throw new Error(`version to upgrade ${versionToUpgradeNumber} is not valid`);
   // temporarily only allow upgrading to the same version as the installed kibana version
-  const kibanaVersion = appContextService.getKibanaVersion();
-  if (kibanaVersion !== version) {
+  if (kibanaVersionNumber !== versionToUpgradeNumber) {
```

Comment:
nitpick the code looks very similar to the one for the bulk, do you think it will make sense to have a function that does that?